static/js/containers/Drawer.js

```diff
@@ -1,105 +1,129 @@
 import type { Location } from "react-router"
 
-// see https://github.com/mitodl/open-discussions/issues/295
-if (!MDCTemporaryDrawer.prototype.getDefaultFoundation_) {
-  MDCTemporaryDrawer.prototype.getDefaultFoundation_ =
-    MDCTemporaryDrawer.prototype.getDefaultFoundation
-  MDCTemporaryDrawer.prototype.getDefaultFoundation = function() {
-    const foundation = this.getDefaultFoundation_()
-
-    foundation.drawerClickHandler_ = e => {
-      if (e.target.tagName !== "A") {
-        e.stopPropagation()
-      }
-    }
-    return foundation
-  }
+export const DRAWER_BREAKPOINT = 950
+
+type DrawerPropsFromState = {
+  showDrawerDesktop: boolean,
+  showDrawerMobile: boolean,
+  subscribedChannels: Array<Channel>
 }
 
-class Drawer extends React.Component<*, void> {
-  // the ref for the rendered DOM element, which MDCTemporaryDrawer needs
-  // access to in order to manage it's animations and so on
-  drawerRoot: HTMLElement | null
-  mdcDrawer: Object
-
-  props: {
-    location: Location,
-    dispatch: Dispatch<*>,
-    showDrawer: boolean,
-    subscribedChannels: Array<Channel>
+type DrawerProps = DrawerPropsFromState & {
+  location: Location,
+  dispatch: Dispatch<*>
+}
+
+export class ResponsiveDrawer extends React.Component<DrawerProps, *> {
+  width: number
+
+  constructor(props: DrawerProps) {
+    super(props)
+    this.width = getViewportWidth()
   }
 
   componentDidMount() {
-    this.mdcDrawer = new MDCTemporaryDrawer(this.drawerRoot)
-    this.mdcDrawer.listen("MDCTemporaryDrawer:close", this.onDrawerClose)
+    window.addEventListener("resize", () => this.onResize())
   }
 
-  onDrawerClose = () => {
-    const { dispatch, showDrawer } = this.props
-
-    // two MDCTemporaryDrawer:close events seem to fire when the drawer
-    // closes, so this is a little hack to debounce that.
-    if (showDrawer) {
-      dispatch(setShowDrawer(false))
+  onResize() {
+    const { showDrawerMobile, dispatch } = this.props
+    const newWidth = getViewportWidth()
+
+    if (
+      newWidth > this.width &&
+      newWidth > DRAWER_BREAKPOINT &&
+      this.width < DRAWER_BREAKPOINT &&
+      showDrawerMobile
+    ) {
+      dispatch(setShowDrawerMobile(false))
     }
-  }
 
-  componentWillUnmount() {
-    if (this.mdcDrawer) {
-      this.mdcDrawer.destroy()
-    }
+    this.width = newWidth
+
+    // this setState call forces a re-render of the component
+    // to ensure that the drawer is responsive
+    this.setState({})
   }
 
-  componentDidUpdate() {
-    const { showDrawer } = this.props
-    this.mdcDrawer.open = showDrawer
+  onDrawerClose = () => {
+    const { dispatch, showDrawerMobile } = this.props
+
+    if (showDrawerMobile) {
+      dispatch(setShowDrawerMobile(false))
+    }
   }
 
   render() {
     const {
+      showDrawerDesktop,
+      showDrawerMobile,
       subscribedChannels,
       location: { pathname }
     } = this.props
+    const isMobile = getViewportWidth() < DRAWER_BREAKPOINT
```

Comment:
[Toolbar.js](https://github.com/mitodl/open-discussions/pull/799/files#diff-821830c49cddb1365a62b0054b6f63ebR43) has the same code to get the viewport width and compare it to the breakpoint. We should create a utility/helper function for this (`isMobileWidth` et al)